### PR TITLE
library: Ported `Menu Button` in rust

### PR DESCRIPTION
**Pull request by mbilal234**
Added menu button implementation in rust.
Preview looks fine, but two errors are seen on the console that I cant seem to understand what they are about
They dont seem to effect the preview or Rust code.
image

Review please!

### CODE REVIEW

File: src/Library/demos/Menu Button/code.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+use crate::workbench;
+use gtk::prelude::*;
+use gtk::{Button, Switch};
+
+pub fn main() {
+    let circular_switch: Switch = workbench::builder()
+        .object("circular_switch")
+        .expect("Failed to get circular_switch");
+    let primary_button: Button S= workbench::builder()
```
Comment (Hofer-Julian) [inline src/Library/demos/Menu Button/code.rs:9]: The S looks broken :)

Comment (mbilal234) [reply src/Library/demos/Menu Button/code.rs]: Check now, pls

File: src/Library/demos/Menu Button/code.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+use gtk::prelude::*;
+use gtk::{Button, Switch};
+
+pub fn main() {
+    let circular_switch: Switch = workbench::builder()
+        .object("circular_switch")
+        .expect("Failed to get circular_switch");
+    let primary_button: Button = workbench::builder()
+        .object("primary")
+        .expect("Failed to get primary_button");
```
Comment (Hofer-Julian) [inline src/Library/demos/Menu Button/code.rs:11]:
```suggestion
```

File: src/Library/demos/Menu Button/code.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    let secondary_button: Button = workbench::builder()
```
Comment (Hofer-Julian) [inline src/Library/demos/Menu Button/code.rs:12]: That should be a `MenuButton` not a `Button`

**Review by Hofer-Julian — COMMENTED** [review]
A couple of small things, then it looks good :)

File: src/Library/demos/Menu Button/code.rs (hunk 3 above)
Comment (mbilal234) [reply src/Library/demos/Menu Button/code.rs]: Check now pls

Comment (Hofer-Julian) [reply src/Library/demos/Menu Button/code.rs]: Have you tried the code?
it doesn't look like it would work. I also don't think it's formatted.

Comment (mbilal234) [reply src/Library/demos/Menu Button/code.rs]: I forgot to add an import, it works on my side now + did the formatting.

Pls check :)

Comment (Hofer-Julian) [reply src/Library/demos/Menu Button/code.rs]: Doesn't look formatted to me :)

Please run the code in Workbench, that will format it.
Then copy the result to `code.rs`.

See here:
https://github.com/sonnyp/Workbench/pull/554#discussion_r1309819779

Comment (Hofer-Julian) [reply src/Library/demos/Menu Button/code.rs]: Also, there are code conflicts now, let me know if you need help with solving them.

Comment (mbilal234) [reply src/Library/demos/Menu Button/code.rs]: The code conflict occured cuz i added my name in About.js, should i remove that?

Comment (mbilal234) [reply src/Library/demos/Menu Button/code.rs]: Also, I ran the code and copy-pasted it as you asked. Let me know if its correctly formatted now

Comment (Hofer-Julian) [reply src/Library/demos/Menu Button/code.rs]: The code looks good now.

you don’t have to remove your name, you should resolve the conflict in a way that all of the additions are there. You can give it a try by clicking on “Resolve conflicts” on this page

Comment (Hofer-Julian) [reply src/Library/demos/Menu Button/code.rs]: I've done it for you.
You can see how in this screencast, so you can try next time on your own :)
[screencast](https://github.com/sonnyp/Workbench/assets/30049909/534cda8b-8b66-411c-8d58-90fee73694f7)

**Comment by Hofer-Julian** [comment]
After your approval @sonnyp, we can merge this

**Pull request merged**